src/components.js

```diff
@@ -75,3 +127,55 @@ export const MagicLinkButton = () => html`
     <button type="submit" style="background: #28a745; width: 100%;">Send Magic Link</button>
   </form>
 `
+
+// Passkey button component  
+export const PasskeyButton = ({ action = 'signin' }) => html`
+  <form method="get" action="${action === 'signup' ? '/signup/passkey' : '/login/passkey'}">
+    <button 
+      type="submit" 
+      style="background: #6f42c1; width: 100%; color: white;"
+    >
+      ${action === 'signup' ? 'Sign Up with Passkey' : 'Sign In with Passkey'}
+    </button>
+  </form>
+`
+
+// Passkey list component
+export const PasskeyList = ({ passkeys }) => html`
+  ${passkeys && passkeys.length > 0 ? html`
```

Comment:
I think you can use the [optional chaining operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining) and make this easier to read. Also, 0 is `falsy`, so we don't need to compare against it to get a truthy value for the ternary operator used.

```suggestion
  ${passkeys?.length ? html`
```